packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -86,19 +95,42 @@ function FlowsheetDataGrid(props) {
     rowHeight,
     onCellSelect,
     onClearSelectedCells,
+    intl,
   } = props;
 
+  // Replace each non-header cell that contains no content with a dash indicating "No results".
+  const parsedRows = rows.map(
+    (row) => ({
+      ...row,
+      cells: row.cells.map(
+        (cell, index) => ((!cell.content && index !== 0) ? {
+          ...cell,
+          content: (
+            <>
+              <span aria-hidden>{intl.formatMessage({ id: 'Terra.flowsheetDataGrid.no-result-display' })}</span>
+              <VisuallyHiddenText text={intl.formatMessage({ id: 'Terra.flowsheetDataGrid.no-result' })} />
+            </>
+          ),
+        } : cell),
```

Comment:
This logic might benefit from the introduction of some variables and an if statement.  The code is logically correct.  It just seems like there is a decent amount of chaining.  For example, maybe there should be variables to store the cloned row and cell objects.  Then you could access the cells and content properties from those objects in your logic.